### DISCUSSION TITLE
What is the easiest way to integrate  social share buttons?

### DISCUSSION THREAD

**Discussion by Djas2017** [Lednerb/bilberry-hugo-theme#317] [Ideas]
Hi,

May be this feature already exists in the bilberry-hugo-theme.
 
If it is not available yet, what is the easiest way to add social sharing buttons to each blog page (gallery, article, etc)?

Some of the third party social sharing solutions: https://www.sharethis.com/ and https://www.addthis.com/

Thanks

**Comment by Lednerb** [comment]
At the moment there are no social share buttons integrated.
You can only link to your accounts in the footer.

I will mark this as a feature request for the upcoming versions.

------------------------

If anyone wants to submit a PR, you should implement an option to activate the sharing buttons via the `config.toml` file.

**Comment by Lednerb** [comment]
We should implement this project because it's privacy protecting and highly configurable:
https://github.com/heiseonline/shariff

**Comment by anoff** [comment]
What do you think about using plain `href` based sharing like

```html
<a href="https://twitter.com/home?status=omg%20look%20at%20this%20awesome%20blog%20post%20I%20found">Share on Twitter</a>
```

see http://www.sharelinkgenerator.com/ for the different formats.

The only privacy issue there is that the social site knows the link is coming from your blog; but that is already in the content. No way for the site to track users on the blog plus no external dependencies added

**Comment by Lednerb** [comment]
@anoff I like this idea, but we would have to style the different, etc. ourselves.

With sharrif, we could simply include the minified CSS and minified JS files (locally) and add a config option to enable/disable it and set the supported platforms.

As I understand the sharrif project, there should not be any tracking.

**Comment by frigi83** [comment]
I have made a PR #221 with direct integration in the theme for sharing sites. There is no need for JS or external CSS. Like social media, I use the Font Awesome icons.

The solution is more like the https://sharingbuttons.io/, but more flexible.

**Comment by KHannibal** [comment]
I'd love to see that feature implemented. I used shariff on WordPress and it was great.
